flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/SourceCodeGenerator.java

```diff
@@ -43,6 +44,87 @@ public String generateCodeSnippetForType(TypeData typeData) {
         };
     }
 
+    public String generateGraphqlClassType(TypeData typeData) {
+        NodeKind nodeKind = typeData.codedata().node();
+        if (nodeKind != NodeKind.CLASS) {
+            return "";
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("\nservice class ")
+                .append(typeData.name())
+                .append(" {");
+
+        // Add inferred fields from functions
+        for (Function function : typeData.functions()) {
+            generateInferredGraphqlClassField(function, stringBuilder);
+        }
+
+        // init functions
+        stringBuilder.append("\n\t").append("function init(");
+        if (!typeData.functions().isEmpty()) {
+            for (int i = 0; i < typeData.functions().size(); i++) {
+                Function function = typeData.functions().get(i);
+                generateTypeDescriptor(function.returnType(), stringBuilder);
+                stringBuilder.append(" ").append(function.name());
+                if (i < typeData.functions().size() - 1) {
+                    stringBuilder.append(", ");
+                }
+            }
+        }
+
+        stringBuilder.append(") {");
+        if (!typeData.functions().isEmpty()) {
+            for (Function function : typeData.functions()) {
+                stringBuilder.append("\n\t\tself.")
+                        .append(function.name())
+                        .append(" = ")
+                        .append(function.name()).append(";");
```

Comment:
Instead of completely depending on StringBuilder with a lot of manual `.append()` calls, we can alternatively use java String formatting or Text Blocks (along with StringBuilder) for better readability and less complexity.